### ISSUE TITLE
Changing source slow and sometimes does not complete.

### ISSUE THREAD

**Issue by lloyda** [johanfrick/custom-sonos-card#5]
If I change the source  (to a favourite) it takes longer than I would expect.  (Compared to mini-media-player) If I then select another favourite, my speakers never seem to complete the move to the new source until I restart HA.  Turning on debug there seems to be a lot of chatter whilst trying to change the source.

**Comment by BrianKarlsen** [comment]
sam here.
Looks like a great card though

**Comment by johanfrick** [comment]
Yes, I agree. I forked this source from https://github.com/exetico/custom-sonos-card and I am not sure what the problem is. 
It looks quite straight forward in the code: https://github.com/johanfrick/custom-sonos-card/blob/master/custom-sonos-card.js#L343
Let me know if you have any ideas!

**Comment by johanfrick** [comment]
@BrianKarlsen @lloyda Please have another go, the fix above should resolve your issues.

**Comment by BrianKarlsen** [comment]
Much better.
Do you know if the playlist will be visible at some point to select a specific track?

**Comment by johanfrick** [comment]
I’m afraid not. I’ve been looking for that as well, but it doesn’t seem like it’s exposed by any api.
